### ISSUE TITLE
Better interface for other devs

### ISSUE THREAD

**Issue by Dummerle** [derrod/legendary#129]
It is very hard to use legendary in my project. A better interface for other developers would be a very nice feature

**Comment by derrod** [comment]
It would be helpful if you could provide specific examples of *what* could be better. Just saying "make it better" is absolutely useless feedback or criticism.

**Comment by Dummerle** [comment]
Some methods in cli.py take just the parameter "args". So it is hard to see, which arguments I need in my function. The list functions in core are good, but the launch and auth functions are not so easy to use. 
I tried to build an easier interface but some functions are hard to implement, so I have not finished.
I hope you understand my problem

**Comment by derrod** [comment]
Nothing in the CLI is supposed to be used by external users. It's just the command line interface for the main legendary project and can change at any time (and will soon enough). In a future version I might look into preventing others from importing the cli module into their code. The core itself is supposed to not change drastically, though I will probably change how the downloading process works quite significantly in a future update.

I know that auth and launch aren't exactly the cleanest, but you'll have to be more specific what about them is hard to use.

**Comment by alx365** [comment]
IMO the interface to devs is amazing! I made a gui frontend for this with a widgeting system, just from shellscript oneliners and the `--json` flag.
![image](https://user-images.githubusercontent.com/30902201/97797203-19ef4680-1c13-11eb-9974-e5fb4873c8a9.png)
(i wanted to share this.. and this is the perfect issue to do so)

**Comment by salva09** [comment]
Why not creating an API? Then making a GUI would be easier

**Comment by derrod** [comment]
There is an API.

**Comment by salva09** [comment]
Yeah, I just've seen it, sorry. Can it be used only in a python script or can be used by other languages? I'm also interested in the GUI because I don't want to type "legendary launch *a bunch of numers*" every time I want to play something xd

**Comment by derrod** [comment]
Scripts yes, if legendary is installed as a python package you can just `from legendary.core import LegendaryCore` for example and use its features. Other languages? No, you'd be better of re-implementing things anyway.

**Comment by StefanLobbenmeier** [comment]
@salva09 can you specify which language you were planning on using? E.g. for Java and JavaScript you should be able to use GraalVM for polyglot projects.

Otherwise from python I agree that the import solution from @derrod and the json solution @alx365 proposed should be good enough and is the most we can expect from this project.

**Comment by salva09** [comment]
> @salva09 can you specify which language you were planning on using? E.g. for Java and JavaScript you should be able to use GraalVM for polyglot projects.
> 
> Otherwise from python I agree that the import solution from @derrod and the json solution @alx365 proposed should be good enough and is the most we can expect from this project.

I wanted to use Rust, but it really is better to use python

**Comment by gpioneer90** [comment]
When a GUI is implemented, may I suggest it be optional and that you could still launch games with CLI

**Comment by theCrius** [comment]
Sorry to barge in, just discovered this project.

Is there an "official" or "community chosen" UI project/feature going on for this project.
While the CLI is perfect, I can see it being a nuisance for non-tech people.

I've good experience with angular/vue and electron but if there is something already going I'd like to take a peek of the work done until now and see if I can help.

**Comment by djey47** [comment]
Hello, no UI project for this still? Could not agree more with @theCrius!

> Sorry to barge in, just discovered this project.
> 
> Is there an "official" or "community chosen" UI project/feature going on for this project?
> While the CLI is perfect, I can see it being a nuisance for non-tech people.
> 
> I've good experience with angular/vue and electron but if there is something already going I'd like to take a peek of the work done until now and see if I can help.

**Comment by StefanLobbenmeier** [comment]
I guess it would be nice if this project would officially promote https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher
